src/components/SearchForm/SearchForm.spec.tsx

```diff
@@ -5,25 +5,26 @@ import userEvent from "@testing-library/user-event";
 import { SearchForm } from "./SearchForm";
 import { ButtonTexts, InputPlaceholders } from "@shared/components";
 
+const mockUpdateSearchParams = jest.fn();
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useOutletContext: jest.fn(() => ({
+    updateSearchParams: mockUpdateSearchParams,
+  })),
+}));
+
 describe("SearchForm", () => {
-  it("renders with initial search query", () => {
-    render(
-      <SearchForm
-        initialSearchQuery="Initial Query"
-        onSearchClick={jest.fn()}
-      />,
-    );
+  it("should be rendered with initial search query", () => {
+    render(<SearchForm initialSearchQuery="Initial Query" />);
```

Comment:
Looks good. Just if you want to do some more stuff, you can use some components form ant design, like pagination and add some pagination in your app, do not show 2000 items at the same time. you pagination shpuld also be in the url `...sortBy=title&page=1` like this. And also search and sort should work. If you want

Reply:
Thanks a lot for suggestion, good experience. Please check